packages/swc-configs/tests/dev.test.ts

```diff
@@ -1,56 +1,115 @@
+test("when fastRefresh is true, react refresh is true", () => {
     const result = defineDevConfig({
+        browsers: Browsers,
         fastRefresh: true
     });
 
-    expect(result).toMatchSnapshot();
+    expect(result.jsc?.transform?.react?.refresh).toBeTruthy();
+});
+
+test("when fastRefresh is false, react refresh is false", () => {
+    const result = defineDevConfig({
+        browsers: Browsers,
+        fastRefresh: false
+    });
+
+    expect(result.jsc?.transform?.react?.refresh).toBeFalsy();
 });
 
-test("when \"parser\" is \"ecmascript\", the configuration parser is ecmascript", () => {
+test("when parser is \"ecmascript\", the configuration parser is ecmascript", () => {
     const result = defineDevConfig({
+        browsers: Browsers,
         parser: "ecmascript"
     });
 
-    expect(result).toMatchSnapshot();
+    expect(result.jsc?.parser?.syntax).toBe("ecmascript");
 });
 
-test("when a config override function is provided, the function argument is the config with the non-config override options applied", () => {
-    const expectedArgument = defineDevConfig({
-        fastRefresh: true
+test("when parser is \"ecmascript\", jsx parsing is enabled", () => {
+    const result = defineDevConfig({
+        browsers: Browsers,
+        parser: "ecmascript"
     });
 
-    const fct = jest.fn<Config, [Config]>(() => ({
-        jsc: {
-            parser: {
-                syntax: "ecmascript"
-            }
-        }
-    }));
+    expect((result.jsc?.parser as EsParserConfig).jsx).toBeTruthy();
+});
 
-    defineDevConfig({
-        fastRefresh: true,
-        configOverride: fct
+test("when parser is \"typescript\", the configuration parser is typescript", () => {
+    const result = defineDevConfig({
+        browsers: Browsers,
+        parser: "typescript"
     });
 
-    expect(fct).toHaveBeenCalledWith(expectedArgument);
+    expect(result.jsc?.parser?.syntax).toBe("typescript");
 });
 
-test("providing options doesn't alter the default config object", () => {
-    expect(DefaultDevConfig.jsc.parser.syntax).toBe("typescript");
+test("when parser is \"typescript\", tsx parsing is enabled", () => {
+    const result = defineDevConfig({
+        browsers: Browsers,
+        parser: "typescript"
+    });
 
-    defineDevConfig({
-        parser: "ecmascript"
+    expect((result.jsc?.parser as TsParserConfig).tsx).toBeTruthy();
+});
+
+test("when a transformer is provided, the transformer is applied on the swc config", () => {
+    const minifyTransformer: SwcConfigTransformer = (config: Config) => {
+        config.minify = true;
+
+        return config;
+    };
+
+    const result = defineDevConfig({
+        browsers: Browsers,
+        transformers: [minifyTransformer]
+    });
+
+    expect(result.minify).toBeTruthy();
+});
+
+test("when multiple transformers are provided, all the transformers are applied on the swc config", () => {
```

Comment:
should we add a testt that validates that the transformer are ran sequentially? in case a transformer depends on another

Reply:
I would greatly warn against having transformers depending on each others so I don't think we should do this.

Furthermore, if someone is willing to do this, this dependency should be embedded in the dependend transformers rather than expecting the transformer feature to garantee ordering.